contentcuration/contentcuration/frontend/shared/views/ImmersiveModalLayout.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+
+  <VApp>
+    <VToolbar app dark color="primary">
+      <VBtn v-if="backButton" flat link :to="previousPage">
+        <Icon>arrow_back</Icon>
+        <span class="back-text">{{ appBarText }}</span>
+      </VBtn>
+
+      <template v-else>
+        <VBtn link icon dark :to="previousPage">
+          <Icon>close</Icon>
+        </VBtn>
+        <VToolbarTitle>
+          {{ appBarText }}
+        </VToolbarTitle>
+      </template>
+    </VToolbar>
+    <VContent>
+      <div class="content-wrapper">
+        <PageContainer class="content">
+          <slot></slot>
+        </PageContainer>
+      </div>
+    </VContent>
+  </VApp>
+
+</template>
+
+
+<script>
+
+  import PageContainer from './PageContainer';
+
+  export default {
+    name: 'ImmersiveModalLayout',
+    components: {
+      PageContainer,
+    },
+    props: {
+      previousPage: {
+        type: [String, Object],
```

Comment:
Could you add a comment or something to say that Object = router link data?

Reply:
done